File: gestaolegal/models/notificacao.py
from dataclasses import dataclass
from datetime import date, datetime
from typing import Literal

TipoNotificacao = Literal["caso", "evento", "lembrete", "plantao"]


@dataclass
class Notificacao:
    acao: str
    data: date
    id_executor_acao: int | None = None
    id_usu_notificar: int | None = None  # None = aviso geral
    tipo: str | None = None
    id_caso: int | None = None
    id_referencia: int | None = None
    lida: bool = False
    data_criacao: datetime | None = None
    id: int | None = None


@dataclass
class NotificacaoListItem(Notificacao):
    executor: str | None = None

File: web/src/lib/types/notificacao.ts
/** Notificação interna (módulo "Notificações"). */
export type Notificacao = {
	id: number;
	acao: string;
	data: string;
	data_criacao: string | null;
	id_executor_acao: number | null;
	/** Nulo = aviso geral (abertura do plantão). */
	id_usu_notificar: number | null;
	tipo: 'caso' | 'evento' | 'lembrete' | 'plantao' | null;
	id_caso: number | null;
	id_referencia: number | null;
	lida: boolean;
	/** Nome de quem executou a ação. */
	executor: string | null;
};

/** Página de destino da notificação, conforme o tipo. */
export function destinoDaNotificacao(n: Notificacao): string | null {
	switch (n.tipo) {
		case 'caso':
			return n.id_caso ? `/casos/${n.id_caso}` : null;
		case 'evento':
			return n.id_caso && n.id_referencia ? `/casos/${n.id_caso}/eventos/${n.id_referencia}` : null;
		case 'lembrete':
			return n.id_caso ? `/casos/${n.id_caso}` : null;
		case 'plantao':
			return '/plantao/escala';
		default:
			return null;
	}
}
